bcipy/signal/model/evaluate/fusion.py

```diff
@@ -0,0 +1,370 @@
+        test_data_eeg = eeg_inquiries[:, test_indices, :]
+        eeg_inquiry_timing = np.array(eeg_inquiry_timing)
+        train_eeg_inquiry_timing = eeg_inquiry_timing[train_indices]
+        test_eeg_inquiry_timing = eeg_inquiry_timing[test_indices]
+        inquiry_symbols_test = np.array([])
+        for t_i in test_indices:
+            inquiry_symbols_test = np.append(inquiry_symbols_test, inquiry_symbols[t_i * 10:(t_i + 1) * 10])
+        inquiry_symbols_test = inquiry_symbols_test.tolist()
+
+        # Now extract the inquiries from trials for eeg model fitting:
+        preprocessed_train_eeg = eeg_model.reshaper.extract_trials(
+            train_data_eeg, trial_duration_samples, train_eeg_inquiry_timing)
+        preprocessed_test_eeg = eeg_model.reshaper.extract_trials(
+            test_data_eeg, trial_duration_samples, test_eeg_inquiry_timing)
+
+        # train and save the eeg model a pkl file
+        # Flatten the labels (0=nontarget/1=target) prior to model fitting
+        erp_train_labels = eeg_inquiry_labels[train_indices].flatten().tolist()
+        # erp_test_labels = eeg_inquiry_labels[test_indices].flatten().tolist()
+        eeg_model.fit(preprocessed_train_eeg, erp_train_labels)
+        eeg_model.metadata = SignalModelMetadata(device_spec=device_spec_eeg,
+                                                 transform=default_transform)
+        # save_model(eeg_model, Path(data_folder, f"model_{eeg_model.auc:0.4f}.pkl"))
+        preferences.signal_model_directory = data_folder
+
+        # extract train and test indices for gaze data:
+        centralized_gaze_data_train = centralized_gaze_data[train_indices]
+        gaze_train_labels = np.array([target_symbols[i] for i in train_indices])
+        gaze_data_test = preprocessed_gaze_data[test_indices]         # test set is NOT centralized
+        gaze_test_labels = np.array([target_symbols[i] for i in test_indices])
+        # generate a tuple that matches the index of the symbol with the symbol itself:
+        symbol_to_index = {symbol: i for i, symbol in enumerate(symbol_set)}
+
+        # train and save the gaze model as a pkl file:
+        reshaped_data = centralized_gaze_data_train.reshape((len(centralized_gaze_data_train), inquiry_length * 4))
+        units = 1e4
+        reshaped_data *= units
+        cov_matrix = np.cov(reshaped_data, rowvar=False)
+        time_horizon = 9
+
+        for eye_coord_0 in range(4):
+            for eye_coord_1 in range(4):
+                for time_0 in range(inquiry_length):
+                    for time_1 in range(inquiry_length):
+                        l_ind = eye_coord_0 * inquiry_length + time_0
+                        m_ind = eye_coord_1 * inquiry_length + time_1
+                        if np.abs(time_1 - time_0) > time_horizon:
+                            cov_matrix[l_ind, m_ind] = 0
+
+        reshaped_mean = np.mean(reshaped_data, axis=0)
+
+        # eps = 5e+3
```

Comment:
?

Reply:
If you are asking about the `reshaped_mean`,  it is the sample average of reshaped inquiry data. As for `eps`, that was a small epsilon value added to the covariance matrix, as a workaround to non-invertibility. But I think we solved that problem.